src/main.ts

```diff
@@ -29,7 +30,12 @@ if (!IsPost) {
   run(inputs)
   // Publish a variable so that when the POST action runs, it can determine it should run the post logic.
   // This is necessary since we don't have a separate entry point
-  coreCommand.issueCommand('save-state', {name: 'isPost'}, 'true')
+  let GH_STATE_FILE = ''
+  if (process?.env?.GITHUB_STATE) {
+    //appendFileSync only accepts string, not 'undefined'.
+    GH_STATE_FILE = process.env.GITHUB_STATE
+  }
+  fs.appendFileSync(GH_STATE_FILE, `isPost=true${os.EOL}`, {encoding: 'utf8'})
```

Comment:
There is a more idiomatic way of doing this documented here https://github.com/actions/toolkit/tree/main/packages/core#action-state.
You may need to update `@actions/core` for using it.

 ```suggestion
  core.saveState("isPost", 'true');
```

Reply:
right, will update dependencies with appropriate version and update the build.